R/utils-get_code_dependency.R

```diff
@@ -49,9 +49,9 @@ get_code_dependency <- function(code, names, check_names = TRUE) {
   graph <- code_graph(calls_pd)
   ind <- unlist(lapply(names, function(x) graph_parser(x, graph)))
 
-  se_ind <- default_side_effects(calls_pd)
+  lib_ind <- detect_libraries(calls_pd)
 
-  as.character(code[unique(c(se_ind, ind))])
+  as.character(code[unique(c(lib_ind, ind))])
```

Comment:
I think this operation could be implemented in `teal`. If anyone adds `library` twice to the `qenv` then we shouldn't fix this I guess. If someone combines `library`, `detach` and `library` again, we shouldn't remove duplicate.

Problem of duplicated libraries is something we create in `teal` by adding `library` calls in `get_rcode_header`

Reply:
I'm not sure if I understand. 

> I think this operation could be implemented in teal.

You mean `library` calls detection should be handled by functions from https://github.com/insightsengineering/teal/issues/950 ?

> If anyone adds library twice to the qenv then we shouldn't fix this I guess. If someone combines library, detach and library again, we shouldn't remove duplicate.

We can introduce one more step that checks if a `library` is called more than once on the very same package, and if it is, detect how many detaches there were and figure out how many times we need to skip `library()` calls. I imagine someone can call `library(package1)` N times and `detach(package1)` M times. We can handle that, even though I would make a user tag `detach()` with `# @linksto` tags.

Reply:
Yeah, I would also leave it to the user to mess up the code as they please